plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/RewriteVarcharConstant.java

```diff
@@ -48,12 +48,13 @@ public Optional<String> rewrite(Constant constant, Captures captures, RewriteCon
             return Optional.empty();
         }
 
-        boolean isAscii = CharMatcher.ascii().matchesAllOf(slice.toStringUtf8());
+        String sliceUtf8String = slice.toStringUtf8();
+        boolean isNonAscii = !CharMatcher.ascii().matchesAllOf(sliceUtf8String);
 
```

Comment:
`CharMatcher.ascii().negate().precomputed().matchesAnyOf(sliceUtf8String)`

Reply:
Move CharMatcher instance to a const.

Reply:
And let's rename it to isUnicodeString

Reply:
fixed

Reply:
I looked at implementation, and looks like it's not working as we expect in more performance way( like as soon it will find non ascii char, it stops processing and return), instead it just use negotiation of match all :(

Reply:
Sorry, can't find it, I see only nonstatic method slice.toStringAscii()